state/state.go

```diff
@@ -399,15 +399,16 @@ func (state *HelmState) UpdateDeps(helm helmexec.Interface) []error {
 // 	 be constructed relative to the `base path`.
 // - Everything else is assumed to be an absolute path or an actual <repository>/<chart> reference.
 func normalizeChart(basePath, chart string) string {
-	if !isLocalChart(chart) {
+	regex, _ := regexp.Compile("^[.]?./")
+	if !regex.MatchString(chart) {
 		return chart
 	}
 	return filepath.Join(basePath, chart)
 }
 
 func isLocalChart(chart string) bool {
-	regex, _ := regexp.Compile("^[.]?./")
-	return regex.MatchString(chart)
+	_, err := os.Stat(chart)
```

Comment:
Just curious but doesn't this result in `chart: stable/mysql` without `./` to be considered a local chart when a directory named `stable/mysql` exists? My example isn't practical but it still seems slightly better to leave the regex matching along with the new file existence check you've added.

Reply:
I don't really agree as the regex matching is limited and does not support all situations.  Like fully qualified paths.

Reply:
That's also true!

Certainly I'm not aware of every case. But anyway, do we need to support fully qualified paths?

To me, it seems like the opposite of helmfile's primary motivation of declaratility :thinking:

Reply:
In that case why support in local files?  My only comment on this is a file existence check is less error prone compared to the regex.  Unless we used a protocol like...

`chart: stable/blah` means a chart in a repository
`chart: file://stable/blah` means a local file.

Reply:
The protocol is interesting!

Anyway, local charts are declarative enough to me as I can normally spot
which version of local chart is used by seeing e.g. git commit's id. I can't expect that for absolute paths, right?

The existence check is definitely nice! I just thought that, assuming we'd need support for relative but absolute path, we can have both, or no necessity to avoid - regex + existence check :thinking:

Reply:
absolute paths are certainly just as viable as relative paths if not more so.  If things are running in a docker container and you are mounting a directory or if you are using a path such as `/opt/blah`

I would lean toward the removal of the regex and using the file existance.

Reply:
Makes sense! Thx.